.github/workflows/4.run-matrixed-pytest-dagger.yml

```diff
@@ -0,0 +1,23 @@
+name: run-matrixed-pytest-dagger
+on: [push]
+jobs:
+  run-matrixed-pytest-dagger:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10","3.11"]
+    steps:
+      - name: Checkout repo content
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: 1.4
+      - name: Install dependencies
+        run: poetry install
+      - name: Run Dagger using Python SDK
+        run: poetry run python dagger/run_tests.py ${{ matrix.python-version }}
```

Comment:
My understanding of the matrix strategy is that it runs multiple jobs concurrently, if resources permit; in this case, I imagine it'd run one job for Python 3.10 and one for 3.11. Your code in `dagger/run_tests.py` leads me to believe that it's supposed to receive multiple versions as an argument and run multiple dagger tests within a single job, but as far as I can tell it'd only ever receive one version. Perhaps it'd make sense to replace the matrix strategy with a variable that enumerates the Python versions to test, then pass that as a whole to `run_tests.py`?

Feel free to disregard if the intent was in fact to use the matrix strategy, but just within dagger rather than running pytest directly.

Reply:
Thanks - this was intentional and maybe should be spelled out a bit better within the file or elsewhere. Using Github Actions in this way with the additional specificity from Dagger tasks allows the actions to span out on Github without losing the reproducibility of Dagger. The default concurrent job within a local environment uses Dagger's suggested concurrency, but may not take full advantage of the Github platform. In practice I've found that when using multiple Python version concurrent tasks from Dagger without Github matrix strategies can sometimes result in halted/incomplete jobs, I believe due to resource constraints. It also has the side effect of portraying the Github Actions workflow more visually, making the results I feel more observable.

Reply:
I see; that makes sense, yeah. So, just to check that I understand, I'm assuming that you allow the user to specify multiple versions as arguments to `dagger/run_tests.py` for the case when they're running locally and want to test against multiple Python versions. When running as a GH action you just give it one version, which still makes use of Dagger's predictable environment. If that's correct, then this is fine, and I don't think any additional explanation is required.